core/trino-main/src/test/java/io/trino/operator/output/BenchmarkPartitionedOutputOperator.java

```diff
@@ -318,10 +323,45 @@ public OptionalInt getNullChannel()
                 }
             },
             LONG_DECIMAL(createDecimalType(MAX_SHORT_PRECISION + 1), 5000),
+            DICTIONARY_LONG_DECIMAL(createDecimalType(MAX_SHORT_PRECISION + 1), 5000) {
+                @Override
+                public Page createPage(List<Type> types, int positionCount, float nullRate)
+                {
+                    return createRandomDictionaryPage(types, positionCount, nullRate);
+                }
+            },
             INTEGER(IntegerType.INTEGER, 5000),
+            DICTIONARY_INTEGER(IntegerType.INTEGER, 5000) {
+                @Override
+                public Page createPage(List<Type> types, int positionCount, float nullRate)
+                {
+                    return createRandomDictionaryPage(types, positionCount, nullRate);
+                }
+            },
             SMALLINT(SmallintType.SMALLINT, 5000),
+            DICTIONARY_SMALLINT(SmallintType.SMALLINT, 5000) {
+                @Override
+                public Page createPage(List<Type> types, int positionCount, float nullRate)
+                {
+                    return createRandomDictionaryPage(types, positionCount, nullRate);
+                }
+            },
             BOOLEAN(BooleanType.BOOLEAN, 5000),
+            DICTIONARY_BOOLEAN(BooleanType.BOOLEAN, 5000) {
+                @Override
+                public Page createPage(List<Type> types, int positionCount, float nullRate)
+                {
+                    return createRandomDictionaryPage(types, positionCount, nullRate);
+                }
+            },
             VARCHAR(VarcharType.VARCHAR, 5000),
+            DICTIONARY_VARCHAR(VarcharType.VARCHAR, 5000) {
+                @Override
+                public Page createPage(List<Type> types, int positionCount, float nullRate)
```

Comment:
nit: it would be better to use composition instead of overriding `createPage`. Right now you pass some arguments to `TestType`, but you also optionally override a method. With composition, you could reuse `createRandomDictionaryPage` rather than override it every time. Composition is also cleaner and more consistent than inheritance

Reply:
makes sense. I refactored it to use new `PageGenerator` interface